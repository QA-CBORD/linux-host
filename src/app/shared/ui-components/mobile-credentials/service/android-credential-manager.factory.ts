import { Injectable } from '@angular/core';
import { LoadingService } from '@core/service/loading/loading.service';
import { AlertController, ModalController, PopoverController, ToastController } from '@ionic/angular';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { AndroidCredential, GoogleCredential } from '../model/android/android-credentials';
import { HidCredentialDataService } from './hid-credential-data.service';
import { HIDCredentialManager } from '../model/android/hid/hid-credential-manager';
import { MobileCredentialManager } from '../model/shared/mobile-credential-manager';
import { GooglePayCredentialManager } from '../model/android/google-pay/google-pay-credential-manager';

@Injectable({
  providedIn: 'root',
})
export class AndroidCredentialManagerFactory {
  constructor(
    private readonly modalCtrl: ModalController,
    private readonly alertCtrl: AlertController,
    private readonly popoverCtrl: PopoverController,
    private readonly loadingService: LoadingService,
    private readonly toastService: ToastController,
    protected readonly hidCredentialDataService: HidCredentialDataService
  ) {}

  getCredentialManager(): Observable<MobileCredentialManager> {
    return this.hidCredentialDataService.androidActivePassesFromServer().pipe(
      map((mobileCredential: AndroidCredential<any>) => {
        let androidCredentialManager = null;
        if (mobileCredential.isHID()) {
          androidCredentialManager = this.createHidCredentialManager();
          androidCredentialManager.setCredential(mobileCredential);
        } else if (mobileCredential.isGoogle()) {
          androidCredentialManager = this.createGoogleCredentialManager(); // google/nxp credential manager not implemented yet
          androidCredentialManager.setCredential(mobileCredential);
        }
        return androidCredentialManager;
      })
    );
  }

  private createHidCredentialManager(): HIDCredentialManager {
    return HIDCredentialManager.getInstance(
      this.modalCtrl,
      this.alertCtrl,
      this.popoverCtrl,
      this.toastService,
      this.loadingService,
      this.hidCredentialDataService
    );
  }

  private createGoogleCredentialManager(): GooglePayCredentialManager {
    return new GooglePayCredentialManager();
  }
}
