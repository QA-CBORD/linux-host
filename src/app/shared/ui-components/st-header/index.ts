export * from './st-header.component';
export * from './st-header.module';
