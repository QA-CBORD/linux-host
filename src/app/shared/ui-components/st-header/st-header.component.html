<ion-header class="st-header">
    <ion-toolbar mode="ios" class="st-header__nav-toolbar">
        <ion-buttons slot="start">
            <ion-back-button
                    class="st-header__back-btn"
                    color="dark"
                    [text]="backButtonTitle"
                    mode="ios"
            ></ion-back-button>
        </ion-buttons>
        <ion-title class="st-header__title">{{ title }}</ion-title>
    </ion-toolbar>

    <ion-toolbar class="st-header__toolbar">
        <ion-searchbar
                (ionChange)="onInputChanged($event)"
                placeholder="Filter Locations..."
                mode="ios"
                class="st-header__searchbar"
        ></ion-searchbar>
    </ion-toolbar>
</ion-header>
