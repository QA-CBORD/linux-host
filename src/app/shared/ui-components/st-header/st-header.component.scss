@import 'tools';

.st-header {
  &.header-md:after {
    background-image: none;
  }

  ion-back-button {
    display: block;
  }

  &__nav-toolbar {
    --min-height: 27px;
    --padding-bottom: 0;
    --padding-top: 0;

    @include bp-grid-desktop {
      display: none;
    }
  }

  &__toolbar {
    --padding-bottom: 0;
    --padding-top: 0;

    &:last-child {
      --border-width: 0;
    }
  }

  &__searchbar {
    --background: #f0f3f5 !important;
    --placeholder-color: #000;
    --icon-color: #c4c4c4 !important;

    padding-top: 0;
    padding-bottom: 0;
    height: 50px;
  }

  &__back-btn {
    --icon-font-size: 18px;
    --margin-start: 10px;
    --icon-padding-end: 5px;

    @include font-nunito-regular(16px);
  }

  &__title {
    @include font-nunito-heavy(20px);
  }
}
