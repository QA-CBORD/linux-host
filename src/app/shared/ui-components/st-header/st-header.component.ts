import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
import { Keyboard } from '@capacitor/keyboard';

@Component({
  selector: 'st-header',
  templateUrl: './st-header.component.html',
  styleUrls: ['./st-header.component.scss'],
})
export class StHeaderComponent {
  @Input() title: string;
  @Input() placeholder: string;
  @Input() backButtonTitle = 'Back';
  @Input() backButtonIcon: string | null = null;
  @Input() isToolbarShow = false;
  @Input() isSubToolbarShow = false;
  @Input() isTitleShow = false;
  @Output() onSearchedValue = new EventEmitter<string>();
  @Input() isBackButtonShow = true;
  @Input() isDismissButtonShow: boolean;
  @Input() isRemoveButtonShow: boolean;
  @Output() onDismiss = new EventEmitter<void>();
  @Output() onRemove = new EventEmitter<void>();

  onInputChanged(event) {
    this.onSearchedValue.emit(event.target.value);
  }

  onEnterKeyClicked() {
    Keyboard.hide();
  }

  onDissmissClicked() {
    this.onDismiss.emit();
  }

  onRemoveClicked() {
    this.onRemove.emit();
  }
}
