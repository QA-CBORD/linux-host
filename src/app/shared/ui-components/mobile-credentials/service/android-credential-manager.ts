import { HttpClient } from '@angular/common/http';
import { ChangeDetectorRef, Injectable } from '@angular/core';
import { AuthFacadeService } from '@core/facades/auth/auth.facade.service';
import { ContentStringsFacadeService } from '@core/facades/content-strings/content-strings.facade.service';
import { InstitutionFacadeService } from '@core/facades/institution/institution.facade.service';
import { LoadingService } from '@core/service/loading/loading.service';
import { PartnerPaymentApiService } from '@core/service/payments-api/partner-payment-api.service';
import { StorageStateService } from '@core/states/storage/storage-state.service';
import { AlertController, ModalController, PopoverController } from '@ionic/angular';
import { from, Observable, of } from 'rxjs';
import { map, switchMap, take } from 'rxjs/operators';
import { AbstractAndroidCredentialDataService } from '../model/android/abstract-android-credential-data.service';
import { AndroidCredential } from '../model/android/android-credentials';
import { HIDCredentialManager } from '../model/android/hid/hid-credential-manager';
import { MobileCredential } from '../model/shared/mobile-credential';
import { CredentialStateChangeSubscription, MobileCredentialManager } from '../model/shared/mobile-credential-manager';



@Injectable({
  providedIn: 'root',
})
export class AndroidCredentialManager extends AbstractAndroidCredentialDataService
  implements MobileCredentialManager {
  private mCredentialManager: MobileCredentialManager;

  constructor(
    private readonly modalCtrl: ModalController,
    private readonly alertCtrl: AlertController,
    private readonly popoverCtrl: PopoverController,
    private readonly loadingService: LoadingService,
    protected partnerPaymentApi: PartnerPaymentApiService,
    protected readonly storageStateService: StorageStateService,
    protected readonly authFacadeService: AuthFacadeService,
    protected readonly institutionFacadeService: InstitutionFacadeService,
    protected readonly contentStringFacade: ContentStringsFacadeService,
    protected readonly httpClient: HttpClient
  ) {
    super(partnerPaymentApi, storageStateService, authFacadeService, institutionFacadeService, httpClient);
  }

  setCredentialStateChangeSubscrption(credentialStateChangeSubscription: CredentialStateChangeSubscription): void {
    this.mCredentialManager.setCredentialStateChangeSubscrption(credentialStateChangeSubscription);
  }

  onUiIconClicked(): void {
    this.mCredentialManager.onUiIconClicked();
  }


  getCredential(): MobileCredential {
    return this.mCredentialManager.getCredential();
  }

  private initManager(): Observable<boolean> {
    return this.androidActivePassesFromServer().pipe(
      map(mCredential => {
        let androidCredential = <AndroidCredential<any>>mCredential;
        if (androidCredential.isHID()) {
            this.mCredentialManager = HIDCredentialManager.getInstance(
            this.modalCtrl,
            this.alertCtrl,
            this.popoverCtrl,
            this.loadingService,
            this.partnerPaymentApi,
            this.storageStateService,
            this.authFacadeService,
            this.institutionFacadeService,
            this.contentStringFacade,
            this.httpClient
          );
          this.mCredentialManager.setCredential(androidCredential);
          return true;
        } else if (androidCredential.isGOOGLE()) {
          this.mCredentialManager = null; // google/nxp credential manager not implemented yet
          this.mCredentialManager.setCredential(androidCredential);
          return true;
        }
        return false;
      })
    );
  }

setCredential(mobileCredential: MobileCredential): void {
    this.mCredentialManager.setCredential(mobileCredential);
}

  credentialEnabled$(): Observable<boolean> {
    return this.initManager().pipe(
        switchMap(userIsEntittled => {
          if(this.mCredentialManager && userIsEntittled) {
             return this.mCredentialManager.credentialEnabled$();
          }
          return of(false);
        })
      );
  }


  initialize(): Promise<any> {
    return this.mCredentialManager.initialize();
  }
  onUiImageClicked(event?: any): void {
    this.mCredentialManager.onUiImageClicked();
  }

  credentialAvailable$(): Observable<boolean> {
    return this.mCredentialManager ? this.mCredentialManager.credentialAvailable$(): of(false);
  }
}
