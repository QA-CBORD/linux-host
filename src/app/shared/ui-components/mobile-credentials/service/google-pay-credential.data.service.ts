import { Injectable } from '@angular/core';
import { AuthFacadeService } from '@core/facades/auth/auth.facade.service';
import { ContentStringsFacadeService } from '@core/facades/content-strings/content-strings.facade.service';
import { InstitutionFacadeService } from '@core/facades/institution/institution.facade.service';
import { APIService } from '@core/service/api-service/api.service';
import { StorageStateService } from '@core/states/storage/storage-state.service';
import { Observable, of } from 'rxjs';
import { GOOGLE, GoogleCredential } from '../model/android/android-credential.model';
import { AndroidCredentialDataService } from '../model/shared/android-credential-data.service';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { map } from 'rxjs/operators';

const major_version = 1,
  minor_version = 0;
const resourceUrls = {
  credentialUrl: `/mf2go/cbord/${major_version}/${minor_version}/credential`,
  ping: `/mf2go/testing/${major_version}/${minor_version}/ping`,
};

const extraHeaders = {
  'x-api-key': 'uIBXQVw9ATQlkn1t0hk91xUvulJo9xS5PjRsU6lh',
};

@Injectable()
export class GooglePayCredentialDataService extends AndroidCredentialDataService {
  constructor(
    protected readonly storageStateService: StorageStateService,
    protected readonly authFacadeService: AuthFacadeService,
    protected readonly contentStringFacade: ContentStringsFacadeService,
    protected readonly institutionFacadeService: InstitutionFacadeService,
    protected readonly apiService: APIService,
    protected readonly http: HttpClient
  ) {
    super(
      resourceUrls,
      storageStateService,
      authFacadeService,
      contentStringFacade,
      institutionFacadeService,
      apiService,
      http
    );
  }

  androidCredential$(reqBody: {
    referenceIdentifier: string;
    googlePayNonce: string;
    otherOptions?: object;
  }): Observable<GOOGLE> {
    return super.androidCredential$(reqBody, extraHeaders);
  }

  protected getDefaultHeaders(): Observable<HttpHeaders> {
    return of(extraHeaders).pipe(
      map(headers => {
        return new HttpHeaders({
          ...headers,
        });
      })
    );
  }

  updateCredential$(mCredential: GoogleCredential): Observable<any> {
    let requestBody = {
      referenceIdentifier: mCredential.getReferenceIdentifier(),
      status: mCredential.getCredStatus(),
      credentialID: mCredential.getId(),
    };

    return super.updateCredential$(requestBody);
  }
}
