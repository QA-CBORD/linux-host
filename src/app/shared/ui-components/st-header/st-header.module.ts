import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { IonicModule } from '@ionic/angular';
import { StHeaderComponent } from './st-header.component';

const declarations = [StHeaderComponent];

@NgModule({
  declarations,
  imports: [
    CommonModule,
    IonicModule
  ],
  exports: declarations
})
export class StHeaderModule { }
