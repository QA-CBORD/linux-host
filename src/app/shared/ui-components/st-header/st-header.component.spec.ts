import { HttpClientTestingModule } from '@angular/common/http/testing';
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ModalController, PopoverController } from '@ionic/angular';
import { StHeaderComponent } from './st-header.component';
import { NativeProvider } from '@core/provider/native-provider/native.provider';
import { Storage } from '@ionic/storage';
import { MockStorageService } from '@core/states/storage/storage-state.service.mock';

const modalController = {
  create: jest.fn(),
};

const popoverCtrl = {};

const nativeProvider = {};

describe('StHeaderComponent', () => {
  let component: StHeaderComponent;
  let fixture: ComponentFixture<StHeaderComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [HttpClientTestingModule],
      declarations: [StHeaderComponent],
      providers: [
        { provide: ModalController, useValue: modalController },
        { provide: PopoverController, useValue: popoverCtrl },
        { provide: NativeProvider, useValue: nativeProvider },
        { provide: Storage , useClass: MockStorageService }
      ],
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(StHeaderComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create the about page', () => {
    expect(component).toBeTruthy();
  });

  it('should dismiss the modal', () => {
    const spy = jest.spyOn(component.onDismiss, 'emit');
    component.onDissmissClicked();
    expect(spy).toHaveBeenCalledTimes(1);
  });
});
