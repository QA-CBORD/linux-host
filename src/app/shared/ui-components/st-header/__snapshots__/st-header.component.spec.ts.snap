// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StHeaderComponent should match the snapshot 1`] = `
<div
  id="root1"
>
  <ion-header
    class="st-header"
    no-border=""
  >
    
    
  </ion-header>
</div>
`;
